### DISCUSSION TITLE
Secure message specification

### DISCUSSION THREAD

**Discussion by dahoo** [cossacklabs/themis#920] [Q&A]
Thank you for the awesome work!

I would like to sign a message with an EC key and verify the signature on a system without themis. For that I am missing some details. Is the signature calculated on a hash of the message? Can I get only the signature or how can I extract the signature without the prepended message?

I guess that would be listed on the specification page: https://docs.cossacklabs.com/themis/spec/secure-message/
Can you estimate when this page will be available or could you share some details in the meantime? I'd appreciate your help.

**Comment by vixentael** [comment]
Hi @dahoo!

May I know why using Themis is not an option on that system? Currently, updating Specs is not our priority, but if I understand your use case better, I will think of smth.

**Reply by dahoo** [reply]
It's a large C++ application which already uses a library that can verify ECDSA signatures, so the developers would prefer to use that instead of adding a new library for that specific use case. The other part is a React Native app, so we were happy to find Themis as good RN crypto libraries are hard to find. 
I would like to know how the signed message and encryption containers look like, so we can extract the parts we need to verify and decrypt in Chilkat.
